### PR TITLE
add multigraph support to igraph_isomorphic

### LINKED ISSUE #2123: igraph_isomorphic() should support non-simple graphs
`igraph_isomorphic()` errors on non-simple graphs.

When it detects such graphs, it should use `igraph_simplify_and_colorize()` in conjunction with VF2 to do the test.

I see `igraph_isomorphic()` as a high-level function that automatically chooses the best method for the given graph. Thus it's fine to add sophisticated decision making like this to it.

### PR DESCRIPTION
**Pull request by GroteGnoom**
fixes #2123

### CODE REVIEW

**Review by ntamas — APPROVED** [review]

**Comment by ntamas** [comment]
This is great, thanks! Can this go on `master` instead of `develop`? `igraph_simplify_and_colorize()` seems to be on `master` already.

Also, can you update the changelog before I merge?

**Comment by szhorvat** [comment]
I'd like to take a look as well before merging.

**Review by szhorvat — APPROVED** [review]

**Comment by szhorvat** [comment]
Thanks @GroteGnoom!

> Can this go on master instead of develop?

IMO yes.

File: src/isomorphism/queries.c [hunk 1]
```diff
@@ -69,15 +69,13 @@ static igraph_error_t igraph_i_isomorphic_small(
...
- * with self-loops, but does not support multigraphs.
- *
  * </para><para>This function decides which graph isomorphism algorithm to be
  * used based on the input graphs. Right now it does the following:
  * \olist
  * \oli If one graph is directed and the other undirected then an
  *    error is triggered.
- * \oli If one of the graphs has multi-edges then an error is triggered.
+ * \oli If one of the graphs has multi-edges then both graphs are
+ *    simplified and colorized and sent to VF2.
```
Comment (szhorvat) [inline src/isomorphism/queries.c:78]:
```suggestion
 *    simplified and colorized using \ref igraph_simplify_and_colorize() and sent to VF2.
```

**Pull request merged**